### PR TITLE
define init sequence in preparation for Go 1.21

### PR DESCRIPTION
**Pull request by briskt**
### Changed
- Perform all package initialization from `main`. This avoids any possible bugs introduced by the new initialization sequence in Go 1.21.

### CODE REVIEW

File: application/domain/domain.go [hunk 1]
```diff
@@ -233,25 +232,28 @@ func init() {
...
-	Env.PremiumMinimum *= CurrencyFactor
-	Env.RepairThresholdString = fmt.Sprintf("%.2g%%", Env.RepairThreshold*100)
-	Env.DeductibleMinimumString = fmt.Sprintf("%.2g%%", Env.Deductible*100)
+	checkSamlConfig(env)
+
+	env.PolicyMaxCoverage *= CurrencyFactor
+	env.DependentAutoApproveMax *= CurrencyFactor
+	env.PremiumMinimum *= CurrencyFactor
+	env.RepairThresholdString = fmt.Sprintf("%.2g%%", env.RepairThreshold*100)
+	env.DeductibleMinimumString = fmt.Sprintf("%.2g%%", env.Deductible*100)
```
Comment (jason-jackson) [inline application/domain/domain.go:249]:
```suggestion
	env.RepairThresholdString = fmt.Sprintf("%.2g%%", env.RepairThreshold*CurrencyFactor)
	env.DeductibleMinimumString = fmt.Sprintf("%.2g%%", env.Deductible*CurrencyFactor)
```

File: application/domain/domain.go [hunk 2]
```diff
@@ -130,8 +129,10 @@ const EnvStaging = "staging"
 const EnvTest = "test"
 
 // Env Holds the values of environment variables
-var Env struct {
-	GoEnv                      string `ignored:"true"`
+var Env = readEnv()
+
+type EnvStruct struct {
+	GoEnv                      string `default:"development" split_words:"true"`
```
Comment (jason-jackson) [inline application/domain/domain.go:135]: Do you know if it's `GOENV` or `GO_ENV`.  I was thinking it was the former, but we were using the latter?

**Review by jason-jackson — APPROVED** [review]

File: application/domain/domain.go (hunk 2 above)
Comment (jason-jackson) [reply application/domain/domain.go:135]: My thoughts were related to `GOOS`/`GOARCH`.  `GO_ENV` is a Buffalo thing.

Comment (briskt) [reply application/domain/domain.go:135]: It's `GO_ENV`. It's not a Go thing (like `GOOS`), from what I can recall. It's only defined in Buffalo.

**Pull request merged**